investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace TRINV.Domain.Validations;
+
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+public static class Ensure
+{
+    public static void IsArgumentNullOrEmpty(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            throw new ArgumentException("The argument value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsArgumentNullOrWhiteSpace(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+        {
+            throw new ArgumentException("The argument value can't be null or white space", paramName);
+        }
+    }
+
+    public static void IsNull<T>(T? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (value is null)
+        {
+            throw new ArgumentException("Object value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsEqual<T>(T? value1, T? value2)
+    {
+        var value1Type = value1.GetType();
```

Comment:
We can simplify this code here in IsEqual. 
Get the type just one time .
The type of the both objects will be the same. The same for the properties.
And you don't need to compare the propertyName . We need just ther value.

Here check this one: More simplified way: 

` 

        Type type = typeof(T);
        foreach (var field in type.GetFields())
        {
            object value1 = field.GetValue(obj1);
            object value2 = field.GetValue(obj2);

            if (value1 == null && value2 == null) continue;
            if (value1 == null || !value1.Equals(value2)) return false;
        }
        return true;

`